tests/MiddleMailServiceTests.cs

```diff
@@ -107,6 +141,27 @@ public async Task ProcessingContinuesAfterGeneralProcessingException() {
 			}
 		}
 
+		[Fact]
+		public async Task RateLimitAllowsCorrectNumberThrough() {
+			for (int i = 0; i < 3; i++) {
+				var emailMessage = FakerFactory.EmailMessageFaker.Generate();
+				await rateLimitedCallback(emailMessage);
+				rateLimitedProcessorMock.Verify(m => m.ProcessAsync(It.IsAny<EmailMessage>()), Times.Exactly(i + 1));
+			}
+
+			// If you want to test that it takes the correct amount of time to delay,
+			// uncomment each commented line,
+			// and change the number of verified sent emails to 4.
+			// DateTime before = DateTime.Now;
+			var notSentMessage = FakerFactory.EmailMessageFaker.Generate();
+			await Task.WhenAny(rateLimitedCallback(notSentMessage), Task.Delay(TimeSpan.FromSeconds(5)));
+			Assert.True(TimeSpan.FromSeconds(5) < MiddleMailService.RateLimitWindow);
+			// await rateLimitedCallback(notSentMessage);
+			// DateTime after = DateTime.Now;
+			// Assert.True(after - before > MiddleMailService.RateLimitWindow);
+			rateLimitedProcessorMock.Verify(m => m.ProcessAsync(It.IsAny<EmailMessage>()), Times.Exactly(3)); //Not sent
+		}
+
```

Comment:
I'm open to suggestions for a better way to do this test, if any reviewer has any such suggestions. That said, both the commented-out version and the currently uncommented version pass.